src/mlir/generator.cc

```diff
@@ -621,6 +652,75 @@ namespace mlir::verona
     return ReturnValue();
   }
 
+  llvm::Expected<ReturnValue> Generator::parseForLoop(const ::ast::Ast& ast)
+  {
+    assert(isFor(ast) && "Bad node");
+
+    // For loops are of the shape (item in list), which need to initialise
+    // the item, take the next from the list and exit if there is none.
+    // All of that is within the scope of the loop, so we need to create a
+    // scope now, to drop them from future ops.
+    SymbolScopeT var_scope{symbolTable};
+
+    // First, we identify the iterator. Nested loops will have their own
+    // iterators, of the same name, but within their own lexical blocks.
+    auto seqNode = getLoopSeq(ast).lock();
+    auto list = parseNode(seqNode);
+    if (auto err = list.takeError())
+      return std::move(err);
```

Comment:
Do you need `std::move` here, can't it be elided?

Reply:
This is the idiom, repeated throughout the code. `err` is an `Error`, the return value is an `Expected`, I believe the elision only happens when the types are the same or a sub-type of, which is not the case.

Reply:
Thanks, I just remember having 
```
warning: moving a local object in a return statement prevents copy elision
```
messages with similar code.